src/appleseed/renderer/kernel/lighting/pt/ptlightingengine.cpp

```diff
@@ -98,6 +98,9 @@ namespace
             const bool      m_enable_caustics;              // are caustics enabled?
 
             const size_t    m_max_path_length;              // maximum path length, ~0 for unlimited
+            const size_t    m_specular_max_path_length;     // maximum specular path length, ~0 for unlimited
```

Comment:
The names of these new properties are not quite accurate as bounces of a given type may not be consecutive.

For instance, in a diffuse -> specular -> diffuse -> specular sequence, there are two diffuse and two specular bounces. So it's not about the maximum length of e.g. specular (sub)paths, but the maximum total number of each bounce type in a given path.

I suggest `m_max_specular_bounces`, etc.